unikraft/app/kernel.go

```diff
@@ -0,0 +1,388 @@
+	libName         string
+	comment         string
+	version         string
+	license         string
+	gitDesc         string
+	ukVersion       string
+	ukFullVersion   string
+	ukCodeName      string
+	ukConfig        []byte
+	ukConfiggz      []byte
+	compiler        string
+	compileDate     string
+	compiledBy      string
+	compiledByAssoc string
+	compileFlags    []string
+}
+
+func libraryFromRecord(libRecord componentInfoRecord) (*lib.LibraryConfig, error) {
+	version := "unknown"
+	if libRecord.version != "" {
+		version = libRecord.version
+	}
+
+	if libRecord.ukVersion != "" {
+		version = libRecord.ukVersion
+	}
+
+	if libRecord.ukFullVersion != "" {
+		version = libRecord.ukFullVersion
+	}
+
+	// map the compile flags slice to a make.ConditionalValue slice
+	cFlags := make([]*makefile.ConditionalValue, 0)
+
+	for _, flag := range libRecord.compileFlags {
+		cFlags = append(cFlags, &makefile.ConditionalValue{
+			DependsOn: nil,
+			Value:     flag,
+		})
+	}
+
+	return lib.NewLibraryConfigFromOptions(lib.WithName(libRecord.libName),
+		lib.WithVersion(version),
+		lib.WithLicense(libRecord.license),
+		lib.WithCompiler(libRecord.compiler),
+		lib.WithCompileDate(libRecord.compileDate),
+		lib.WithCompiledBy(libRecord.compiledBy),
+		lib.WithCompiledByAssoc(libRecord.compiledByAssoc),
+		lib.WithCFlags(cFlags),
+	)
+}
+
+func gunzipConfig(configgz []byte) ([]byte, error) {
+	buf := bytes.NewBuffer(configgz)
+	gz, err := gzip.NewReader(buf)
+	if err != nil {
+		return nil, err
+	}
+	defer gz.Close()
+
+	configData, err := io.ReadAll(gz)
+	if err != nil {
+		return nil, err
+	}
+
+	return configData, nil
+}
+
+func unikraftFromRecord(ctx context.Context, unikraftRecord componentInfoRecord) (*core.UnikraftConfig, error) {
+	var configData []byte
+	var err error
+
+	if unikraftRecord.ukConfig != nil {
+		configData = unikraftRecord.ukConfig
+	} else if unikraftRecord.ukConfiggz != nil {
+		configData, err = gunzipConfig(unikraftRecord.ukConfiggz)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	config, err := kconfig.ParseConfigData(configData)
+	if err != nil {
+		return nil, err
+	}
+
+	configMap := kconfig.KeyValueMap{}
+
+	for _, cfg := range config.Slice {
+		if cfg.Value == kconfig.No {
+			continue
+		}
+		configMap[cfg.Key] = &kconfig.KeyValue{
+			Key:   cfg.Key,
+			Value: cfg.Value,
+		}
+	}
+
+	version := "Unknown"
+	if unikraftRecord.ukVersion != "" {
+		version = unikraftRecord.ukVersion
+	}
+
+	if unikraftRecord.ukFullVersion != "" {
+		version = unikraftRecord.ukFullVersion
+	}
+
+	return core.NewUnikraftFromOptions(ctx, core.WithVersion(version), core.WithKConfig(configMap))
+}
+
+func addFieldToRecord(record *componentInfoRecord, tnum int, data []byte, byteorder binary.ByteOrder, printMetadata bool, configFile string) error {
+	switch tnum {
+	case 0x0001:
+		record.libName = string(data[:len(data)-1])
+		if printMetadata {
+			fmt.Printf("Name:                 %s\n", record.libName)
+		}
```

Comment:
Can you please remove all `printMetadata` if-statements and the `printMetadata` argument from `addFieldToRecord`; this is exposed to us by the result, thus letting the user decide how to format and print information.